### ISSUE TITLE
Support PERMISSION_DENIED_APP_OP

### ISSUE THREAD

**Issue by hotchemi** [hotchemi/PermissionsDispatcher#45]
When the app's compileSdkVersion is under 23, onRequestPermissionResult always return GRANTED by default.
So we should modify to check with PermissionChecker.checkSelfPermissions in onRequestPermissionResult when compileSdkVersion < 23.

**Comment by hotchemi** [comment]
Just lough idea. Not exact.
```diff
public static void onRequestPermissionsResult(MainActivity target, int requestCode, int[] grantResults) {
    switch (requestCode) {
      case REQUEST_SHOWCAMERA:
+      if (PermissionChecker.checkSelfPermission(grantResults) == PackageManager.PERMISSION_DENIED_APP_OP) {
+        target.deniedActionForCamera();
+        return;
+     } 
      if (PermissionUtils.verifyPermissions(grantResults)) {
        target.showCamera();
      } else {
        target.deniedActionForCamera();
      }
      break;
      case REQUEST_SHOWCONTACTS:
      if (PermissionUtils.verifyPermissions(grantResults)) {
        target.showContacts();
      } else {
        target.deniedActionForContacts();
      }
      break;
      default:
      break;
    }
  }
}
```

**Comment by aurae** [comment]
Does there need to be a distinction between PERMISSION_DENIED and PERMISSION_DENIED_APP_OPS? At the moment, PermissionUtils checks if a permission result is "!= GRANTED", is that not enough? What does the APP_OPS constant even do? The docs don't really help.

**Comment by hotchemi** [comment]
@aurae Sorry I fixed the diff. It's enough.
And when it comes to APP_OPS, yup this is not on the android developers...

When:
- App's targetSdkVersion is under 23 and device os is 6.0
If:
- User revoke the permission from setting `grantResults` always return
Then:
- PermissionChecker.checkSelfPermission returns APP_OPS
- `grantResults` of onRequestPermissionsResult is always "GRANTED"

So we have to check again if the targetSdkVersion is under 23...

**Comment by aurae** [comment]
Ohh I see! In that case, I would suggest not altering the generated code, but adding those specifications to `PermissionUtils`.

**Comment by hotchemi** [comment]
I found the problem from this [article](http://blog.riskfinder.co.jp/2015/11/checkselfpermission.html) but this is Japanese...
I guess this requirements is a bug so gonna report to b.android.com :)

**Comment by hotchemi** [comment]
I'm gonna implement this feature!

**Comment by hotchemi** [comment]
Closed by #72 .
